src/metalperformanceshaders.cs

```diff
@@ -4203,159 +6066,2016 @@ interface MPSImageReduceUnary {
+
+		[Export ("primaryScale")]
+		float PrimaryScale { get; set; }
+
+		[Export ("secondaryScale")]
+		float SecondaryScale { get; set; }
+
+		[Export ("bias")]
+		float Bias { get; set; }
+
+		[Export ("primaryStrideInFeatureChannels")]
+		nuint PrimaryStrideInFeatureChannels { get; set; }
+
+		[Export ("secondaryStrideInFeatureChannels")]
+		nuint SecondaryStrideInFeatureChannels { get; set; }
+
+		[Export ("minimumValue")]
+		float MinimumValue { get; set; }
+
+		[Export ("maximumValue")]
+		float MaximumValue { get; set; }
+
+		[Export ("encodeToCommandBuffer:primaryImage:secondaryImage:destinationState:destinationImage:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage primaryImage, MPSImage secondaryImage, MPSCnnArithmeticGradientState destinationState, MPSImage destinationImage);
+
+		[Export ("encodeBatchToCommandBuffer:primaryImages:secondaryImages:destinationStates:destinationImages:")]
+		void EncodeBatch (IMTLCommandBuffer commandBuffer, NSArray<MPSImage> primaryImages, NSArray<MPSImage> secondaryImages, MPSCnnArithmeticGradientState [] destinationStates, NSArray<MPSImage> destinationImages);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnArithmetic), Name = "MPSCNNAdd")]
+	[DisableDefaultCtor]
+	interface MPSCnnAdd {
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnArithmetic), Name = "MPSCNNSubtract")]
+	[DisableDefaultCtor]
+	interface MPSCnnSubtract {
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnArithmetic), Name = "MPSCNNMultiply")]
+	[DisableDefaultCtor]
+	interface MPSCnnMultiply {
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnArithmetic), Name = "MPSCNNDivide")]
+	[DisableDefaultCtor]
+	interface MPSCnnDivide {
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (12,1), Mac (10,14,1, onlyOn64: true), iOS (12,1)]
+	[BaseType (typeof (MPSCnnArithmetic))]
+	[DisableDefaultCtor]
+	interface MPSNNCompare {
+
+		[Export ("comparisonType", ArgumentSemantic.Assign)]
+		MPSNNComparisonType ComparisonType { get; set; }
+
+		[Export ("threshold")]
+		float Threshold { get; set; }
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnGradientKernel), Name = "MPSCNNArithmeticGradient")]
+	[DisableDefaultCtor]
+	interface MPSCnnArithmeticGradient {
+
+		// Inlined but you are not supposed to use this class, only its subclasses
```

Comment:
Same: protected.

Reply:
I would normally agree but subclasses already inline the constructor that is available for them and it is not always both of these.